src/main.tsx

```diff
@@ -35,7 +35,7 @@ const theme = createTheme({
   },
 });
 
-ReactDOM.createRoot(document.getElementById("root") as HTMLElement).render(
+ReactDOM.createRoot(document.getElementById("root")!).render(
```

Comment:
The reason `createRoot` complained is because you better hand it a real element, not `undefined`.  If you know this is going to be the case, and `document.getElementById("root")` will never be null, you can use the `!` or [non-null assertion operator](https://stackoverflow.com/q/42273853/1366033)